### PR TITLE
Implemented metrics public api

### PR DESCRIPTION
**Pull request by disa6302**
*Issue #, if available: #578 *
*Description of changes:*
- Set up basic implementations for public RTC metrics APIs. 
- Upcoming changes:
    - ICE Metrics
    - Filing up some of the portions of public APIs 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/source/Metrics/Metrics.c [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+ * Kinesis WebRTC Metrics
+ */
+#define LOG_CLASS "Metrics"
+#include "../Include_i.h"
+
+STATUS getIceCandidatePairStats(PRtcPeerConnection pRtcPeerConnection, PRtcIceCandidatePairStats pRtcIceCandidatePairStats)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    PKvsPeerConnection pKvsPeerConnection = (PKvsPeerConnection) pRtcPeerConnection;
+    CHK((pRtcPeerConnection != NULL || pRtcIceCandidatePairStats != NULL), STATUS_NULL_ARG);
```
Comment (MushMal) [inline src/source/Metrics/Metrics.c:11]: No need for the internal parenthesis

File: src/source/Metrics/Metrics.c [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+    DLOGW("getIceCandidatePairStats not supported currently");
```
Comment (MushMal) [inline src/source/Metrics/Metrics.c:12]: Also, add CHK(FALSE, STATUS_NOT_IMPLEMENTED);

Comment (MushMal) [reply src/source/Metrics/Metrics.c:12]: Same comments apply to the cases below

File: src/source/Metrics/Metrics.c [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+        case  RTC_STATS_TYPE_DATA_CHANNEL:
+        case  RTC_STATS_TYPE_RECEIVER:
+        case  RTC_STATS_TYPE_SENDER:
+        case  RTC_STATS_TYPE_TRACK:
+        case  RTC_STATS_TYPE_CODEC:
+        case  RTC_STATS_TYPE_SCTP_TRANSPORT:
+        case  RTC_STATS_TYPE_TRANSCEIVER:
+        case  RTC_STATS_TYPE_RTC_ALL:
+        default:
+            DLOGW("Unsupported stats type requested");
```
Comment (MushMal) [inline src/source/Metrics/Metrics.c:104]: Same question - should you return STATUS_NOT_IMPLEMENTED error.

File: src/source/Metrics/Metrics.c [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+    }
+CleanUp:
+    return retStatus;
+}
+
+STATUS getSignalingClientStats(SIGNALING_CLIENT_HANDLE signalingClientHandle, PSignalingClientMetrics pSignalingMetrics)
```
Comment (MushMal) [inline src/source/Metrics/Metrics.c:110]: Remove this - it's being implemented in the signaling

File: src/source/Metrics/Metrics.c (hunk 1 above)
Comment (disa6302) [reply src/source/Metrics/Metrics.c:11]: Done

File: src/source/Metrics/Metrics.c (hunk 2 above)
Comment (disa6302) [reply src/source/Metrics/Metrics.c]: Ah! Wasnt aware of this status code. Fixed everywhere

File: src/source/Metrics/Metrics.c (hunk 4 above)
Comment (disa6302) [reply src/source/Metrics/Metrics.c]: Done

File: src/source/Metrics/Metrics.c (hunk 3 above)
Comment (disa6302) [reply src/source/Metrics/Metrics.c]: Yeah. Seems to be apt

**Pull request merged**